mlrun/execution.py

```diff
@@ -893,6 +895,94 @@ def log_model(
         self._update_run()
         return item
 
+    def log_llm_prompt(
+        self,
+        key,
+        prompt_string: Optional[str] = None,
+        prompt_path: Optional[str] = None,
+        prompt_legend: Optional[dict] = None,
+        model_artifact: Union[ModelArtifact, str] = None,
+        generation_configuration: Optional[dict] = None,
+        description: Optional[str] = None,
+        target_path: Optional[str] = None,
+        artifact_path: Optional[str] = None,
+        tag: Optional[str] = None,
+        labels: Optional[Union[list[str], str]] = None,
+        upload: Optional[bool] = None,
+        **kwargs,
+    ) -> LLMPromptArtifact:
+        """Log an LLM prompt artifact and optionally upload it to the artifact store.
+
+        This function allows you to log a prompt artifact for large language model (LLM) usage. Prompts can be defined
+        as a string or by referencing a file path. Optionally, you can link the prompt to a parent model artifact and
+        provide metadata like a prompt legend (e.g., input variable mapping) and generation configuration.
+
+        Examples::
+
+            # Log an inline prompt
+            context.log_llm_prompt(
+                key="qa-prompt",
+                prompt_string="Q: {question}\nA:",
+                model_artifact="model:latest",
+                prompt_legend={"question": "user_input"},
+                generation_configuration={"temperature": 0.7, "max_tokens": 128},
+                tag="latest",
+            )
+
+            # Log from a file
+            project.log_llm_prompt(
+                key="summarization-prompt",
+                prompt_path="prompts/summarize.txt",
+                labels=["task=summarization"],
+            )
+
+        :param key: Unique name of the artifact.
+        :param prompt_string: Raw prompt text as a string. Cannot be used with `prompt_path`.
+        :param prompt_path: Path to a file containing the prompt content. Cannot be used with `prompt_string`.
+        :param prompt_legend: A dictionary that maps variables used in the prompt to their expected context keys.
```

Comment:
hmm... I was actually thinking that this will indicate per template parameter what the type of the object is. I'm not sure this is really meaningful - mapping to context keys. Let's discuss this.

Reply:
Maybe this better : 
`
        :param prompt_legend: A dictionary where each key is a placeholder in the prompt (e.g., {user_name}) 
                              and the value is a description or explanation of what that placeholder represents.
                              Useful for documenting and clarifying dynamic parts of the prompt.
`